docs/scos/dev/glue-api-guides/202204.0/glue-backend-api/how-to-guides/how-to-create-grant-type-parameters.md

```diff
@@ -0,0 +1,134 @@
+
+Integrate authentication following the [Glue API Authentication integration](/docs/scos/dev/technical-enhancement-integration-guides/glue-authentication-integration.html) guide.
+
+Glue provides grant types `password` for a customer, and a user out of the box:
+
+| SPECIFICATION           | PLUGIN           |
+| -------------- | ----------------- |
+| Customer grant type | Spryker\Zed\Oauth\Communication\Plugin\Oauth\CustomerPasswordOauthRequestGrantTypeConfigurationProviderPlugin |
+| User grant type | Spryker\Zed\Oauth\Communication\Plugin\Oauth\UserPasswordOauthRequestGrantTypeConfigurationProviderPlugin |
+
+
+Let's say you have a `Foo` user, and you want to have a new grant type `password` for it. To create the grant type, follow these steps:
+
+1. Create `FooUserPasswordGrantTypeBuilder`.
+   
+```php
+<?php
+
+namespace Pyz\Zed\Oauth\Business\Grant;
+
+use DateInterval;
+use Spryker\Zed\Oauth\Business\Model\League\Grant\GrantTypeBuilderInterface;
+use Spryker\Zed\Oauth\Business\Model\League\Grant\GrantTypeInterface;
+use Spryker\Zed\Oauth\Business\Model\League\Grant\PasswordGrantType;
+use Spryker\Zed\Oauth\Business\Model\League\RepositoryBuilderInterface;
+
+class FooPasswordGrantTypeBuilder implements GrantTypeBuilderInterface
+{
+    public function buildGrant(
+        RepositoryBuilderInterface $repositoryBuilder,
+        DateInterval $refreshTokenTTL
+    ): GrantTypeInterface {
+        $fooUserPasswordGrantType = new PasswordGrantType();
+        $fooUserPasswordGrantType->setUserRepository($repositoryBuilder->createOauthFooUserRepository());
+        $fooUserPasswordGrantType->setRefreshTokenRepository($repositoryBuilder->createRefreshTokenRepository());
+        $fooUserPasswordGrantType->setRefreshTokenTTL($refreshTokenTTL);
+
+        return $fooUserPasswordGrantType;
+    }
+}
+```
+
+2. Create `FooPasswordOauthRequestGrantTypeConfigurationProviderPlugin`.
+
+```php
+<?php
+
+namespace Pyz\Zed\Oauth\Communication\Plugin\Oauth;
+
+use Generated\Shared\Transfer\GlueAuthenticationRequestContextTransfer;
+use Generated\Shared\Transfer\OauthGrantTypeConfigurationTransfer;
+use Generated\Shared\Transfer\OauthRequestTransfer;
+use Pyz\Zed\Oauth\Business\Grant\FooPasswordGrantTypeBuilder;
+use Spryker\Glue\Kernel\AbstractPlugin;
+use Spryker\Zed\Oauth\OauthConfig;
+use Spryker\Zed\OauthExtension\Dependency\Plugin\OauthRequestGrantTypeConfigurationProviderPluginInterface;
+
+class FooPasswordOauthRequestGrantTypeConfigurationProviderPlugin extends AbstractPlugin implements OauthRequestGrantTypeConfigurationProviderPluginInterface
+{
+    protected const GLUE_BACKEND_API_APPLICATION = 'GLUE_BACKEND_API_APPLICATION';
+
+    public function isApplicable(
+        OauthRequestTransfer $oauthRequestTransfer,
+        GlueAuthenticationRequestContextTransfer $glueAuthenticationRequestContextTransfer
+    ): bool {
+        return (
+            $oauthRequestTransfer->getGrantType() === OauthConfig::GRANT_TYPE_PASSWORD &&
```

Comment:
We already have `password` grand type, and the name is booked. Please add a link for https://oauth.net/2/grant-types/ somewhere, and make an example of https://oauth.net/2/grant-types/authorization-code/ instead.